blocks/reward.go

```diff
@@ -0,0 +1,53 @@
+package blocks
+
+import (
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/log"
+)
+
+// RewardConfig defines the configuration options for Spacemesh rewards.
+type RewardConfig struct {
+	BaseReward uint64 `mapstructure:"base-reward"`
+}
+
+type layerRewardsInfo struct {
+	layerID      types.LayerID
+	numProposals uint64
+	feesReward   uint64
+	layerReward  uint64
+	// smesher reward per proposal
+	totalRewardPer uint64
+	// the layer reward part of the smesher reward per proposal
+	layerRewardPer uint64
+}
+
+func (info *layerRewardsInfo) Fields() []log.LoggableField {
```

Comment:
is it a new code? should be using ObjectMarshaller interface if so

we should stop using Fields interface for logging, should probably add a note that somewhere that it is deprecated

Reply:
thanks for the reminder. i've dropped Fields here and for Ballot and Proposal too.